CAP/gap/DerivedMethods.gi

```diff
@@ -154,6 +154,41 @@ AddWithGivenDerivationPairToCAP( UniversalMorphismIntoZeroObject,
 end : CategoryFilter := IsAdditiveCategory,
       Description := "UniversalMorphismIntoZeroObject computing the zero morphism" );
 
+##
+AddWithGivenDerivationPairToCAP( ProjectionInFactorOfFiberProduct,
+        
+  function( diagram, projection_number )
+    local D, diagram_of_equalizer, iota;
+    
+    D := List( diagram, Source );
+    
+    diagram_of_equalizer := List( [ 1 .. Length( D ) ], i -> ProjectionInFactorOfDirectProduct( D, i ) );
+    
+    diagram_of_equalizer := List( [ 1 .. Length( D ) ], i -> PreCompose( diagram_of_equalizer[i], diagram[i] ) );
+    
+    iota := EmbeddingOfEqualizer( diagram_of_equalizer );
+    
+    return PreCompose( iota, ProjectionInFactorOfDirectProduct( D, projection_number ) );
+    
+  end : Description := "ProjectionInFactorOfFiberProduct by composing the embedding of equalizer with the direct product projection" );
+
+##
+AddWithGivenDerivationPairToCAP( UniversalMorphismIntoFiberProduct,
+        
+  function( diagram, tau )
+    local D, diagram_of_equalizer, chi;
+    
+    D := List( diagram, Source );
+    
+    diagram_of_equalizer := List( [ 1 .. Length( D ) ], i -> ProjectionInFactorOfDirectProduct( D, i ) );
+    
+    diagram_of_equalizer := List( [ 1 .. Length( D ) ], i -> PreCompose( diagram_of_equalizer[i], diagram[i] ) );
+    
+    chi := UniversalMorphismIntoDirectProduct( D, tau );
+    
+    return UniversalMorphismIntoEqualizer( diagram_of_equalizer, chi );
```

Comment:
Unfortunately, this derivation of `UniversalMorphismIntoFiberProduct` 
will only work properly in combination with your above given
derivation of `ProjectionInFactorOfFiberProduct`. Imagine that the user
gives his own `ProjectionInFactorOfFiberProduct`, then there is
no guarantuee that the user's fiber product object (which then equals the source of the user's fiber product projection)
coincides with the equalizer object of the diagram that you create here.

To address such interdependencies between derivations,
we introduced the final derivations and used them
for the parallel derivation in the abelian case
(fiber products derived from kernels and direct sums).


However, Sebastian Gutsche and I are currently thinking about
a more flexible way to deal with such interdependencies, so
you can either wait for our new solution to be implemented
or to create your derivation similar to the one in the abelian case.